src/watchmaker/workers/salt.py

```diff
@@ -273,64 +276,43 @@ def _build_salt_formula(self, extract_dir):
-            temp_extract_dir = extract_dir
-            if self.salt_content_path:
-                temp_extract_dir = os.sep.join((
-                    self.working_dir,
-                    'salt-archive'
-                ))
-            self.extract_contents(
-                filepath=salt_content_file,
-                to_directory=temp_extract_dir
-            )
-            if self.salt_content_path:
+            if not self.salt_content_path:
+                self.extract_contents(
+                    filepath=salt_content_file,
+                    to_directory=extract_dir
+                )
+            else:
                 self.log.debug(
                     'Using salt content path: %s',
                     self.salt_content_path
                 )
+                temp_extract_dir = os.sep.join((
+                    self.working_dir, 'salt-archive'))
+                self.extract_contents(
+                    filepath=salt_content_file,
+                    to_directory=temp_extract_dir
+                )
                 salt_content_src = os.sep.join((
                     temp_extract_dir, self.salt_content_path))
-                salt_files_dir = None
-                for (i, path) in enumerate(glob.glob(salt_content_src)):
-                    if os.path.isdir(path):
-                        if i > 0:
-                            msg = 'Found multiple paths matching' \
-                                  ' \'{0}\' in {1}'.format(
-                                      self.salt_content_path,
-                                      self.salt_content)
-                            self.log.critical(msg)
-                            raise WatchmakerException(msg)
-                        salt_files_dir = path
-                self.log.debug('Using extracted content from: %s',
-                               salt_files_dir)
-
-                for subdir in next(os.walk(salt_files_dir))[1]:
-                    if (
-                            not subdir.startswith('.') and
-                            not os.path.exists(
-                                os.sep.join((extract_dir, subdir)))
-                    ):
-                        watchmaker.utils.copytree(
-                            os.sep.join((salt_files_dir, subdir)),
-                            os.sep.join((extract_dir, subdir))
-                        )
+                salt_content_glob = glob.glob(salt_content_src)
+                self.log.debug('salt_content_glob: %s', salt_content_glob)
+                if len(salt_content_glob) > 1:
+                    msg = 'Found multiple paths matching' \
+                          ' \'{0}\' in {1}'.format(
+                              self.salt_content_path,
+                              self.salt_content)
+                    self.log.critical(msg)
+                    raise WatchmakerException(msg)
+                salt_files_dir = salt_content_glob[0]
+
+                watchmaker.utils.copy_subdirectories(
```

Comment:
after this, we still have `temp_extract_dir` hanging around right? make sense to remove it?

Reply:
Thought about doing that, but the `temp_extract_dir` is being created within the `working_dir`. And since that gets removed during the cleanup routine, figured I'd not bother removing it.
https://github.com/plus3it/watchmaker/blob/62400d9b27f6beac5a4af5d89daf7c072429ce87/src/watchmaker/managers/platform.py#L242

Reply:
let me know if that's not a good reason to leave it in there though.

Reply:
no, that makes sense to me. can address later if we realize otherwise...